recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -72,6 +71,9 @@ class OpenTelemetryCppConan(ConanFile):
 
     @property
     def _min_cppstd(self):
+        if Version(self.version) < "1.12.0" and self.options.with_stl:
+            # Old versions require C++17 with STL
```

Comment:
Forgot to mention, this comes from https://github.com/open-telemetry/opentelemetry-cpp/blob/v1.9.1/CMakeLists.txt#L128